File: Find the Winner of the Circular Game.go
type Node struct {
    Next *Node
    Val int
}

func findTheWinner(n int, k int) int {
    head := &Node{Val:1}
    ptr := head
    for i := 2; i < n+1; i++ {
        ptr.Next = &Node{Val:i}
        ptr = ptr.Next
    }
    ptr.Next = head

    prev := ptr
    ptr = head

    count := k-1
    for prev != ptr {
        if count == 0 {
            prev.Next = ptr.Next
            ptr = ptr.Next
            count = k-1
        } else {
            prev, ptr = ptr, ptr.Next
            count--
        }
    }
    return ptr.Val
}

File: Find the Winner of the Circular Game.cpp
class Node {
    public:
    Node(int val, Node* next) : val(val), next(next) {}

    int val;
    Node* next;
};

class Solution {
public:
    int findTheWinner(int n, int k) {
        auto head = new Node(1, nullptr);
        auto ptr = head;
        for (int i = 2; i < n+1; i++) {
            ptr->next = new Node(i, nullptr);
            ptr = ptr->next;
        }
        ptr->next = head;

        auto prev = ptr;
        ptr = head;
        int count = k-1;
        while (prev != ptr) {
            if (count == 0) {
                prev->next = ptr->next;
                ptr = ptr->next;
                count = k-1;
            } else {
                prev = ptr;
                ptr = ptr->next;
                count--;
            }
        }
        return ptr->val;
    }
};
